ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -182,13 +185,14 @@ class IdpLoginScreen extends React.Component<Props, State> {
                 : "authentication.errors.login.title"
             )}
           </Text>
-          <Text style={styles.errorBody}>
-            {I18n.t(
-              errorType === "LOADING_ERROR"
-                ? "authentication.errors.network.body"
-                : "authentication.errors.login.body"
+          {errorType === "LOGIN_ERROR" &&
+            this.state.errorCode && (
+              <Text style={styles.errorBody}>
+                {I18n.t(("authentication.errors.spid.error_" +
```

Comment:
you have no guarantee the incoming error code is mapped into translation keys
I suggest mapping the handled errors and if an error code is not mapped, you should show a generic error message